### ISSUE TITLE
NetCDF: Parallel operation on file opened for non-parallel access

### ISSUE THREAD

**Issue by venugopalrt** [LBL-EESA/TECA#556]
The following error occurred after successful installation of TECA using netcdf lib compiled and installed with --enable-parallel-4 in Ubuntu desktop 20.04

teca_integrated_vapor_transport --input_regex ./ERA51950.nc --x_axis_variable longitude --y_axis_variable latitude --specific_humidity q --wind_u u --wind_v v --ivt ivt --n_threads -1 --verbose --z_axis_variable level --write_ivt 1 --write_ivt_magnitude 1
STATUS: [0:139641852189056] [/home/user/TECA/core/teca_thread_util.cxx:269 4.0.0-10-g2d023c1]
STATUS: threadpool afinity:
   0 : Luval :   0.0   : 0 1 2 3
[0:139641852189056] teca_index_executive::initialize index_initializer_key=number_of_writers number_of_writers=1 index_request_key=writer_id first=0 last=-1 stride=1 block_start=0 block_size=1
[0:139641852189056] teca_index_executive::get_next_request writer_id=0 extent=0, 500, 0, 320, 0, 0
start_time_step = 0
end_time_step = -1
n_time_steps = 1460
n_time_steps_per_file = 128
n_files = 12
n_ranks = 1
rank    first_time_step last_time_step
0       0       1459
file    ranks
0       0,
1       0,
2       0,
3       0,
4       0,
5       0,
6       0,
7       0,
8       0,
9       0,
10      0,
11      0,
ERROR: [0:139641852189056] [/home/user/TECA/io/teca_netcdf_util.cxx:204 4.0.0-10-g2d023c1]
ERROR: Failed to create "IVT_1950-01-01-00Z.nc". NetCDF: Parallel operation on file opened for non-parallel access
ERROR: [0:139641852189056] [/home/user/TECA/io/teca_cf_layout_manager.cxx:96 4.0.0-10-g2d023c1]
ERROR: failed to create file "IVT_%t%.nc"
ERROR: [0:139641852189056] [/home/user/TECA/io/teca_cf_writer.cxx:287 4.0.0-10-g2d023c1]
ERROR: Failed to create file 0

**Comment by burlen** [comment]
it seems that there is another install of NetCDF that is being found during the build, during the run, or both. I will assume that this other install is something you'd like to keep because for instance you have some other tools that depend on it. 

To help you get to the bottom of it try the following:

Run `grep netcdf CMakeCache.txt` in the build directory. Verify that all the paths and files come from the same install. If not the issue is in the build side of things. The following environment settings may help cmake and the shared library loader find the desired install

During build and when run add your netcdf install to PATH, PKG_CONFIG_PATH, and LD_LIBRARY_PATH.  For example:
```
export PATH=/your/netcdf/install/bin:$PATH
export PKG_CONFIG_PATH=/your/netcdf/install/lib/pkgconfig:$PKG_CONFIG_PATH
export LD_LIBRARY_PATH=/your/netcdf/install/lib:$LD_LIBRARY_PATH
```
this might be necessary to get cmake and the shared library loader to find your install.

When trouble  shooting build issues it's necessary to delete all of cmake generated files because they can retain old values. I usually add an rm command to my cmake line to make sure I always remember to do it. For example:
```
rm -rf *; cmake ..
```
here I left out all of the -D options. One must be very careful to be sure you are in the build dir when issuing such a command!

**Comment by burlen** [comment]
If the above don't help then would you mind uploading or email me directly your CMakeCache.txt file? This will be found in the build dir.

**Comment by burlen** [comment]
our documentation had a typo in the netcdf build flags. fixed in 634198ccaab8
